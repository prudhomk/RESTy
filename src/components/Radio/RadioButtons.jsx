/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';


const RadioButtons = ({ method, onChange }) => {
  return (
    <div>

      <input type="radio" id="get" name="method" value="GET" checked={method === 'GET'} onChange={onChange} />
      <label htmlFor="get">GET</label>

      <input type="radio" id="post" name="method" value="POST" checked={method === 'POST'} onChange={onChange} />
      <label htmlFor="post">POST</label>

      <input type="radio" id="put" name="method" value="PUT" checked={method === 'PUT'} onChange={onChange} />
      <label htmlFor="put">PUT</label>

      <input type="radio" id="patch" name="method" value="PATCH" checked={method === 'PATCH'} onChange={onChange} />
      <label htmlFor="patch">PATCH</label>

      <input type="radio" id="delete" name="method" value="DELETE" checked={method === 'DELETE'} onChange={onChange} />
      <label htmlFor="delete">DELETE</label>
                        
    </div>
  );
};

RadioButtons.propTypes = {
  method: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

export default RadioButtons;
