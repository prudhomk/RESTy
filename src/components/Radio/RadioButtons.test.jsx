import React from 'react';
import { render, cleanup } from '@testing-library/react';
import RadioButtons from '../Radio/RadioButtons';

describe('RadioButtons component', () => {
  afterEach(() => cleanup());
  it('renders RadioButtons', () => {
    const { asFragment } = render(<RadioButtons />);
    expect(asFragment()).toMatchSnapshot();
  });
});
