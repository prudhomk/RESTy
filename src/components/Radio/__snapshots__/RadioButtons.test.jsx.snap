// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButtons component renders RadioButtons 1`] = `
<DocumentFragment>
  <div>
    <input
      id="get"
      name="method"
      type="radio"
      value="GET"
    />
    <label
      for="get"
    >
      GET
    </label>
    <input
      id="post"
      name="method"
      type="radio"
      value="POST"
    />
    <label
      for="post"
    >
      POST
    </label>
    <input
      id="put"
      name="method"
      type="radio"
      value="PUT"
    />
    <label
      for="put"
    >
      PUT
    </label>
    <input
      id="patch"
      name="method"
      type="radio"
      value="PATCH"
    />
    <label
      for="patch"
    >
      PATCH
    </label>
    <input
      id="delete"
      name="method"
      type="radio"
      value="DELETE"
    />
    <label
      for="delete"
    >
      DELETE
    </label>
  </div>
</DocumentFragment>
`;
